src/target/riscv/batch.h

```diff
@@ -14,6 +14,116 @@ enum riscv_scan_type {
 	RISCV_SCAN_TYPE_WRITE,
 };
 
+/* These types are used to specify how many JTAG RTI cycles to add after a
+ * scan.
+ */
+enum riscv_scan_delay_class {
+	/* Delay needed for accessing debug module registers: */
+	RISCV_DELAY_BASE,
+	/* Delay for execution of an abstract command: */
+	RISCV_DELAY_ABSTRACT_COMMAND,
+	/* Delay for System Bus read operation: */
+	RISCV_DELAY_SYSBUS_READ,
+	/* Delay for System Bus write operation: */
+	RISCV_DELAY_SYSBUS_WRITE,
+	/* Number of enum elements: */
+	RISCV_N_DELAY_CLASSES
+};
+
+static inline const char *
+riscv_scan_delay_class_name(enum riscv_scan_delay_class delay_class)
+{
+	switch (delay_class) {
+	case RISCV_DELAY_BASE:
+		return "DM access";
+	case RISCV_DELAY_ABSTRACT_COMMAND:
+		return "Abstract Command";
+	case RISCV_DELAY_SYSBUS_READ:
+		return "System Bus read";
+	case RISCV_DELAY_SYSBUS_WRITE:
+		return "System Bus write";
+	case RISCV_N_DELAY_CLASSES:
+		assert(0);
+	}
+	return NULL;
+}
+
+struct riscv_scan_delays {
+	/* The purpose of these delays is to be passed to "jtag_add_runtest()",
+	 * which accepts an "int".
+	 * Therefore, they should be no greater then "INT_MAX".
+	 */
+	unsigned int base_delay;
+	unsigned int ac_delay;
+	unsigned int sb_read_delay;
+	unsigned int sb_write_delay;
+};
+
+static inline unsigned int riscv_scan_get_delay(struct riscv_scan_delays delays,
+		enum riscv_scan_delay_class delay_class)
+{
+	switch (delay_class) {
+	case RISCV_DELAY_BASE:
+		return delays.base_delay;
+	case RISCV_DELAY_ABSTRACT_COMMAND:
+		return delays.ac_delay;
+	case RISCV_DELAY_SYSBUS_READ:
+		return delays.sb_read_delay;
+	case RISCV_DELAY_SYSBUS_WRITE:
+		return delays.sb_write_delay;
+	case RISCV_N_DELAY_CLASSES:
+		assert(0);
+	}
+	return 0;
+}
```

Comment:
Here I'd recommend to pass the structure via const pointer.

I'd also appears reasonable to add `assert(false)` prior to the `return`.

```suggestion
static inline unsigned int riscv_scan_get_delay(const struct riscv_scan_delays *delays,
		enum riscv_scan_delay_class delay_class)
{
	assert(delays);
	switch (delay_class) {
	case RISCV_DELAY_BASE:
		return delays->base_delay;
	case RISCV_DELAY_ABSTRACT_COMMAND:
		return delays->ac_delay;
	case RISCV_DELAY_SYSBUS_READ:
		return delays->sb_read_delay;
	case RISCV_DELAY_SYSBUS_WRITE:
		return delays->sb_write_delay;
	case RISCV_N_DELAY_CLASSES:
		assert(false);
	}
	assert(false);
	return 0;
}
```

Reply:
Addressed.